### ISSUE TITLE
Fix WideString serialization

### ISSUE THREAD

**Issue by GoogleCodeExporter** [theseion/Fuel#192]
```
See senders of #encodeString:
There are a lot of cases where a WideString can be passed as argument: inst 
vars, class names, global names.

```

Original issue reported on code.google.com by `tinchod...@gmail.com` on 25 Feb 2013 at 4:12

**Comment by ilredeitopi** [comment]
Steps to reproduce:
<pre>
cl := Class new setName: 'Foo' asWideString; yourself.
Smalltalk at: cl name put: cl.
FLSerializer serializeToByteArray: (Smalltalk at: cl name)
</pre>

It produces a 'Improper store into indexable object'

**Comment by theseion** [comment]
Added tests that reproduce the issue. A fix is not going to be straight forward because global names are encoded as strings and are written onto the stream directly.

**Comment by marianopeck** [comment]
Yes, this is a known limitation we have since forever. At the time we were thinking about it, we found out that Pharo tools were not even supporting class names with widestring...so it was not even worth to adapt fuel for that. In all Pharo tools now (browser, monticello, class creation, etc etc) seem to work correct with WideStrings class names, then this issue would make sense to fix.

**Comment by theseion** [comment]
During investigation of this issue we discovered that a global binding can be anything since SystemDictionary is a simply key value store. That means that we can't even be sure that we get a String as the class name (we managed to create a class with name 4 [a SmallInteger]).

I'll write a mail to the list shortly to see what the view is on this topic.
